client/modules/IDE/components/SketchList.jsx

```diff
@@ -25,17 +26,21 @@ class SketchList extends React.Component {
     return `p5.js Web Editor | ${this.props.username}'s sketches`;
   }
 
+  renderLoader() {
+    if (this.props.loading) return <Loader />;
+    if (!this.props.loading && this.props.sketches.length === 0) return (<p className="sketches-table__empty">No sketches.</p>);
+    return null;
+  }
+
   render() {
     const username = this.props.username !== undefined ? this.props.username : this.props.user.username;
     return (
       <div className="sketches-table-container">
         <Helmet>
           <title>{this.getSketchesTitle()}</title>
         </Helmet>
-        { this.props.sketches.length === 0 &&
-          <p className="sketches-table__empty">No sketches.</p>
-        }
-        { this.props.sketches.length > 0 &&
+        {this.renderLoader()}
+        {!this.props.loading && this.props.sketches.length > 0 &&
```

Comment:
could you turn `!this.props.loading && this.props.sketches.length > 0` into a helper function?

Reply:
Should I do something like arePages() && <JSX/>

Reply:
I am sorry for delay , I will finish this today.